### ISSUE TITLE
JwtDecoder should not be discoverable

### ISSUE THREAD

**Issue by jgrandja** [jzheaux/spring-security#17]
Currently, `OAuth2ResourceServerConfigurer.getJwtDecoder()` attempts to discover the `JwtDecoder` in the `ApplicationContext`. This capability should be removed in the inital PR. We may add this later.

The `jwkSetUrl` is required and it will be used to create the existing `NimbusJwtDecoderJwkSupport`.

**Comment by jzheaux** [comment]
Agreed. This isn't required to support `jwkSetUrl`, so we can safely remove it from this PR.

@jgrandja, should we add tickets for features like this that we are taking out to add them back in, say, tied to a give PR (maybe expressed via Milestones...?) Might help with clarity about what is going in which PR.

**Comment by jgrandja** [comment]
I don't feel we need to add a ticket to support a `JwtDecoder ` `@Bean`.
We might go with another angle similar to what you had with a `KeyProvider`.
